internal/store/k8s.go

```diff
@@ -216,11 +219,35 @@ func (s *KubeStore) SetKeeperInfo(ctx context.Context, id string, ms *cluster.Ke
 		if err != nil {
 			return fmt.Errorf("failed to get latest version of pod: %v", err)
 		}
-		if result.Annotations == nil {
-			result.Annotations = map[string]string{}
+
+		oldData, err := json.Marshal(result)
+		if err != nil {
+			return err
+		}
+
+		accessor, err := meta.Accessor(result)
+		if err != nil {
+			return err
+		}
+
+		annotations := accessor.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		annotations[util.KubeStatusAnnnotation] = string(msj)
+		accessor.SetAnnotations(annotations)
```

Comment:
I don't understand why this is needed, isn't this doing the same thing at the previous single line?

```
result.Annotations[util.KubeStatusAnnnotation] = string(msj)
```

If so all of this can be summarized to:

```
		oldData, err := json.Marshal(result)
		if err != nil {
			return err
		}

		result.Annotations[util.KubeStatusAnnnotation] = string(msj)

		newData, err := json.Marshal(result)
		if err != nil {
			return err
		}

		patchBytes, err := jsonpatch.CreateMergePatch(oldData, newData)
		if err != nil {
			return err
		}

		_, err = podsClient.Patch(s.podName, types.MergePatchType, patchBytes)
		return err
```

Reply:
You're totally right. That's because I borrowed the code from `kubectl annotate`, which is designed to be a lot more generic, hence the accessor. I'll fix it.